src/Workspaces/CoreTest/GeneratedCodeRecognitionTests.cs

```diff
@@ -51,9 +52,33 @@ private static void TestFileNames(bool assertGenerated, params string[] fileName
         {
             var project = CreateProject();
 
+            var projectWithUserConfiguredGeneratedCodeTrue = project.AddAnalyzerConfigDocument(".editorconfig",
+                SourceText.From(@"
+[*.{cs,vb}]
+generated_code = true
+"), filePath: @"z:\.editorconfig").Project;
+
+            var projectWithUserConfiguredGeneratedCodeFalse = project.AddAnalyzerConfigDocument(".editorconfig",
+                SourceText.From(@"
+[*.{cs,vb}]
+generated_code = false
+"), filePath: @"z:\.editorconfig").Project;
+
             foreach (var fileName in fileNames)
             {
-                var document = project.AddDocument(fileName, "");
+                TestCore(fileName, project, assertGenerated);
+
+                // Verify user configuration always overrides generated code heuristic.
+                if (fileName.EndsWith(".cs") || fileName.EndsWith(".vb"))
+                {
+                    TestCore(fileName, projectWithUserConfiguredGeneratedCodeTrue, assertGenerated: true);
```

Comment:
Want to instead do this as [Theory] that takes true/false/null in?

Reply:
That would require all tests using the helper to unnecessarily add the same inline data (true for .cs and .vb and false for rest of the extensions). Did I understand your suggestion correctly?

Reply:
Oh I didn't realize this was a common test method. I must have misread the diff and thought it was an explicit test. My bad.